### DISCUSSION TITLE
[Feature request] Checking that a subgraph is a clique

### DISCUSSION THREAD

**Discussion by adam2392** [networkx/networkx#7128] [Algorithms]
In many of the algorithms, there is usually a "is_*" algorithm. For example, if I have a graph, `G` and a set of nodes, `nbunch` then I would want to determine if the subgraph forms a clique. However, this is currently not possible within networkx.

I wanted to see if it would be warranted to introduce a public function in the `clique` submodule like:

```
def is_clique(G, nbunch=None):
      # check if G.subgraph(nbunch) is a clique
      # if nbunch is None, nbunch is G.nodes
```

For reference, there are different ways of checking, but online-ppl have already benchmarked stuff and we can just check the number of edges: https://stackoverflow.com/questions/59009712/fastest-way-of-checking-if-a-subgraph-is-a-clique-in-networkx

**Comment by dschult** [comment]
I believe this would be:
`G.subgraph(nbunch).size() == len(G) * (len(G) - 1) // 2`
Could we use some documentation to show people how to do it rather than creating a new function?
Where would that go?

**Reply by adam2392** [reply]
Perhaps in the doc page for `cliques` it could be a short note?

Or in the docstring of the functions regarding cliques?

**Reply by boothby** [reply]
Edge cases abound here.  Sadly, self-loops, directed graphs, and multigraphs each throw a wrench into that approach.  Also, you won't want to mix `G` and `G.subgraph` like that.

**Reply by dschult** [reply]
Right -- the code I wrote above was not good. I edited it to be correct, but could be made more readable.  We should mention in the docs that different situations call for different definitions, but for ordinary simple undirected graphs with no self-loops, this formula can be used.  I think the `cliques` doc-page is a good place for a paragraph on this.
